tracer/src/Datadog.Trace.Tools.Runner/AnalyzeProcessMemoryCommand.cs

```diff
@@ -0,0 +1,65 @@
+﻿// <copyright file="AnalyzeProcessMemoryCommand.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.IO;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Telemetry;
+using Datadog.Trace.Tools.Runner.Checks;
+using Datadog.Trace.Tools.Runner.DumpAnalysis;
+using Spectre.Console;
+using Spectre.Console.Cli;
+using static Datadog.Trace.Tools.Runner.Checks.Resources;
+
+namespace Datadog.Trace.Tools.Runner
+{
+    internal class AnalyzeProcessMemoryCommand : AsyncCommand<AnalyzeProcessMemorySettings>
+    {
+        public override async Task<int> ExecuteAsync(CommandContext context, AnalyzeProcessMemorySettings settings)
+        {
+            AnsiConsole.Record();
+            AnsiConsole.WriteLine("Running checks on process " + settings.Pid);
+
+            var process = ProcessInfo.GetProcessInfo(settings.Pid);
+            try
+            {
+                if (process == null)
+                {
+                    Utils.WriteError("Could not fetch information about target process. Make sure to run the command from an elevated prompt, and check that the pid is correct.");
+                    return 1;
+                }
```

Comment:
Having this check inside the try-catch means it'll try to upload even if `process is null` Maybe need some `#nullable enable` here😉 

```suggestion
            if (process is null)
            {
                Utils.WriteError("Could not fetch information about target process. Make sure to run the command from an elevated prompt, and check that the pid is correct.");
                return 1;
            }

            try
            {
```

Reply:
Great catch! Fixed in 7ba4bb6bbbcb9992e7874a4625f5a783bbf05002. 
❤️ `#enable nullable` ❤️